Pick changes for 8.4.4

tsuoanttila opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/framework/10989)
<!-- Reviewable:end -->

Commit:
Use random string for autocomplete attribute (#10950)

Fixes #7820

tsuoanttila committed:
Fix IE11 tests for BrowserStack environment (#10959)

OlliTietavainenVaadin committed:
Fix NPE check in BindingValidationStatus (#10964)

nittka committed:
Remove parse error when changing from invalid to empty date text (#10771)

Fixes #10673.

tsuoanttila committed:
Update Spring version (#10971)

tsuoanttila committed:
Update CDI version (#10983)

Commit:
Graceful servlet shutdown if no VaadinService is initialized (#10981)

Fixes #10980

Commit:
Update TestBench version

elmot commented:
Reviewed 2 of 2 files at r1, 40 of 40 files at r2, 2 of 2 files at r3, 3 of 3 files at r4, 1 of 1 files at r5, 1 of 1 files at r6, 1 of 1 files at r7, 1 of 1 files at r8.
Review status: :shipit: complete! all files reviewed, all discussions resolved

---



*Comments from [Reviewable](https://reviewable.io/reviews/vaadin/framework/10989)*
<!-- Sent from Reviewable.io -->

tsuoanttila committed:
Fix TabSheet attaching and detaching components (#10988)

This patch reverts the fix #10557 and replaces it with a
proper solution from Grid perspective.

Fixes #10987
Fixes #10985

elmot commented:
Reviewed 7 of 7 files at r9.
Review status: :shipit: complete! all files reviewed, all discussions resolved

---



*Comments from [Reviewable](https://reviewable.io/reviews/vaadin/framework/10989)*
<!-- Sent from Reviewable.io -->
